### ISSUE TITLE
Gap of validity between IdToken + accessToken And sessionToken

### ISSUE THREAD

**Issue by HDaghash** [aws/amazon-cognito-identity-js#471]
Hi guys ,

we are using cognito for login and api gateway to secure our apis ,

but i'm facing an issue , when i hit the apis after a less than hour the api return Invalid or expired token it seems the  sessionToken and the secretAccessKey is expired while the IdToken + accessToken is still valid .

why these gap between the tokens and keys .

please advice !

**Comment by bradennapier** [comment]
I am noticing the same thing and it makes the entire thing unusable - not sure how I am supposed to reliably call our API if it fails afer validating on the client side... refresh every 10 seconds or something? ridiculous

**Update** It would appear that tokens are only valid to api gateway for about 10 minutes before becoming invalid.  That is absolutely crazy and obivously does not match what is documented (60 minutes).

Please fix this...

**Comment by itrestian** [comment]
Yes, tokens vended by STS are valid for up to 1 hr. You would have to implement logic in your app that refreshes them close to the expiry date.
